src/test/java/part2/exercise/CollectorsExercise1.java

```diff
@@ -51,32 +66,102 @@ public int getDuration() {
-
-        // Second option
-        // Collectors.toMap
-        // iterate twice: stream...collect(...).stream()...
-        // TODO
-        throw new UnsupportedOperationException();
+    private static Stream<PersonPositionDuration> personPositionDurationStream(List<Employee> employees) {
+        return employees.stream()
+                .flatMap(e -> e.getJobHistory().stream()
+                        .map(j -> new PersonPositionDuration(e.getPerson(), j.getPosition(), j.getDuration())));
+    }
+
+    // With the longest duration on single job (single iteration)
+    private Map<String, Person> getCoolestByPositionSingleIteration(List<Employee> employees) {
+        return personPositionDurationStream(employees)
+                .collect(groupingBy(
+                        PersonPositionDuration::getPosition,
+                        collectingAndThen(
+                                maxBy(comparing(PersonPositionDuration::getDuration)), p -> p.get().getPerson())));
+    }
+
+    // With the longest duration on single job (double iteration)
+    private Map<String, Person> getCoolestByPositionDoubleIteration(List<Employee> employees) {
+        return personPositionDurationStream(employees)
+                .collect(
+                        toMap(PersonPositionDuration::getPosition, identity(), BinaryOperator.maxBy(comparing(PersonPositionDuration::getDuration))))
+                .entrySet().stream()
+                .collect(toMap(Map.Entry::getKey, e -> e.getValue().getPerson()));
+    }
+
+    private class CoolestByPositionCollector implements Collector<PersonPositionDuration, Map<String, PersonPositionDuration>, Map<String, Person>> {
+
+        final BinaryOperator<PersonPositionDuration> remappingFunction = BinaryOperator.maxBy(Comparator.comparing(PersonPositionDuration::getDuration));
+
+        @Override
+        public Supplier<Map<String, PersonPositionDuration>> supplier() {
+            return HashMap::new;
+        }
+
+        @Override
+        public BiConsumer<Map<String, PersonPositionDuration>, PersonPositionDuration> accumulator() {
+            return (m, p) -> {
+                m.merge(p.getPosition(), p, remappingFunction);
+            };
+        }
+
+        @Override
+        public BinaryOperator<Map<String, PersonPositionDuration>> combiner() {
+            return (a, b) -> {
+                b.entrySet().forEach(e -> a.merge(e.getKey(), e.getValue(), remappingFunction));
+                return a;
+            };
+        }
+
+        @Override
+        public Function<Map<String, PersonPositionDuration>, Map<String, Person>> finisher() {
+            return (m) -> m.entrySet().stream()
+                    .collect(toMap(Map.Entry::getKey, e -> e.getValue().getPerson()));
+        }
+
+        @Override
+        public Set<Characteristics> characteristics() {
+            return Collections.unmodifiableSet(
+                    EnumSet.of(Characteristics.CONCURRENT));

```

Comment:
Incorrect. `CONCURRENT` means that you can add elements into single accumulator from multiple threads. Actually you can't add elements into `HashMap` from multiple threads.